### PR TITLE
CASSSIDECAR-175: Renamed InstancesConfig to InstancesMetadata

### PR DESCRIPTION
**Pull request by nvharikrishna**
Suggesting to refactor InstancesConfig to InstancesMetadata as all the methods in it are returning InstanceMetadata and not InstanceConfiguration.

For CASSANDRASC-175

### CODE REVIEW

**Comment by yifan-c** [comment]
Nice. Another contribution! 

I updated the issue title to include the JIRA id. This way, the PR can be linked back to JIRA.

**Comment by bbotella** [comment]
@yifan-c should it be renamed to [CASSSIDECAR-175](https://issues.apache.org/jira/browse/CASSSIDECAR-175) instead of CASSANDRASC-175?

**Comment by bbotella** [comment]
Thanks a lot for the patch! Taking a look at `InstancesConfig` javadoc:
```
Maintains metadata of instances maintained by Sidecar
```

What do you folks think about being super explicit with the new name with something like `OwnedCassandraInstancesMetadata`?

**Comment by frankgh** [comment]
I think `OwnedCassandraInstancesMetadata` would be too much. I'd prefer clear javadocs stating what the code does

**Comment by bbotella** [comment]
If current naming is consensus, then I'm +1 on the patch (nb). Thanks a lot @nvharikrishna !

**Review by bbotella — APPROVED** [review]

File: server/src/main/java/org/apache/cassandra/sidecar/cluster/InstancesMetadataImpl.java [hunk 1]
```diff
@@ -33,19 +33,19 @@
 /**
  * Local implementation of InstancesConfig.
  */
-public class InstancesConfigImpl implements InstancesConfig
+public class InstancesMetadataImpl implements InstancesMetadata
```
Comment (frankgh) [inline server/src/main/java/org/apache/cassandra/sidecar/cluster/InstancesMetadataImpl.java:36]: update javadoc above as well

File: server/src/main/java/org/apache/cassandra/sidecar/server/MainModule.java [hunk 2]
```diff
@@ -476,14 +476,14 @@ public DriverUtils driverUtils()
...
     @Singleton
-    public InstancesConfig instancesConfig(Vertx vertx,
-                                           SidecarConfiguration configuration,
-                                           CassandraVersionProvider cassandraVersionProvider,
-                                           SidecarVersionProvider sidecarVersionProvider,
-                                           DnsResolver dnsResolver,
-                                           CQLSessionProvider cqlSessionProvider,
-                                           DriverUtils driverUtils,
-                                           MetricRegistryFactory registryProvider)
+    public InstancesMetadata instancesConfig(Vertx vertx,
```
Comment (frankgh) [inline server/src/main/java/org/apache/cassandra/sidecar/server/MainModule.java]: rename the method as well
```suggestion
    public InstancesMetadata instancesMetadata(Vertx vertx,
```

File: server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java [hunk 3]
```diff
@@ -184,21 +184,21 @@ public void setSslConfiguration(SslConfiguration sslConfiguration)
         refreshInstancesConfig();
     }
 
-    public InstancesConfig instancesConfig()
+    public InstancesMetadata instancesConfig()
```
Comment (frankgh) [inline server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java]: rename the method here too?
```suggestion
    public InstancesMetadata instancesMetadata()
```

File: server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java [hunk 4]
```diff
@@ -184,21 +184,21 @@ public void setSslConfiguration(SslConfiguration sslConfiguration)
...
+        if (instancesMetadata == null)
         {
             refreshInstancesConfig();
         }
-        return this.instancesConfig;
+        return this.instancesMetadata;
     }
 
-    public InstancesConfig refreshInstancesConfig()
+    public InstancesMetadata refreshInstancesConfig()
```
Comment (frankgh) [inline server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java]:
```suggestion
    public InstancesMetadata refreshInstancesMetadata()
```

File: server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java [hunk 5]
```diff
@@ -228,23 +228,23 @@ public String toString()
...
         for (InstanceConfigListener listener : instanceConfigListeners)
         {
-            listener.onInstancesConfigChange(this.instancesConfig);
+            listener.onInstancesConfigChange(this.instancesMetadata);
         }
     }
 
-    private InstancesConfig buildInstancesConfig(CassandraVersionProvider versionProvider,
-                                                 DnsResolver dnsResolver)
+    private InstancesMetadata buildInstancesConfig(CassandraVersionProvider versionProvider,
```
Comment (frankgh) [inline server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java]:
```suggestion
    private InstancesMetadata buildInstancesMetadata(CassandraVersionProvider versionProvider,
```

**Review by frankgh — COMMENTED** [review]
Thanks for the patch! We still have a few places where we use the `InstancesConfig` naming, I found a few places, but I'd suggest you go through the codebase to find any that might be missing.

**Comment by nvharikrishna** [comment]
@yifan-c @bbotella @frankgh Thaks for taking a look!

Updated the remaining references of InstancesConfig  to InstancesMetadata.

**Comment by frankgh** [comment]
The PR looks good, but needs rebasing and probably updating after the latest merges

File: server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java (hunk 5 above)
Comment (nvharikrishna) [reply server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java]: done

**Comment by nvharikrishna** [comment]
Rebased the code

**Review by frankgh — APPROVED** [review]
+1 Looks good to me

**Pull request merged**